### PR TITLE
cleanup spec text, update links, example, and references

### PR DESCRIPTION
**Pull request by mkhraisha**
Did a review of the document, cleaned up text, links, examples and references

### CODE REVIEW

File: docs/spec/index.html [hunk 1]
```diff
@@ -236,7 +236,7 @@
     </p>
 
     <p>
-      To contribute to this vocabulary or reference technical details related to
+      To contribute to this specification or reference technical details related to
       the project, please reference the
```
Comment (TallTed) [inline docs/spec/index.html:240]:
```suggestion
      the project, please refer to the
```

File: docs/spec/index.html [hunk 2]
```diff
@@ -345,24 +345,26 @@ <h2>Rules for Processing Data</h2>
...
-        directly.
-      </p>
-      <section class="normative">
+        <section class="normative">
+          <h2>Identifiers</h2>
+          <p>
+            Identifiers in objects exchanged using this specification are
+            particularly important, not least in that they uniquely identify
+            an object or network transaction, but also, given the
+            asynchronous nature of this mode of data exchange, that they
```
Comment (TallTed) [inline docs/spec/index.html:354]:
```suggestion
            asynchronous nature of this mode of data exchange, in that they
```

File: docs/spec/index.html [hunk 3]
```diff
@@ -345,24 +345,26 @@ <h2>Rules for Processing Data</h2>
...
+            present an area where various attacks could arise by sending
```
Comment (TallTed) [inline docs/spec/index.html:355]:
```suggestion
            present an area where various attacks could send
```

File: docs/spec/index.html [hunk 4]
```diff
@@ -345,24 +345,26 @@ <h2>Rules for Processing Data</h2>
...
+            data with identifiers already in use with malicious intent to
+            confuse a receiver of the data.
+          </p>
+          <p>
+            Identifiers conformant with this spec <em>MUST</em> be
```
Comment (TallTed) [inline docs/spec/index.html:360]:
```suggestion
            Identifiers conformant with this specification <em>MUST</em> be
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Editorial language fixes

File: docs/spec/index.html (hunk 3 above)
Comment (TallTed) [reply docs/spec/index.html:355]: or
```suggestion
            present an area where various attacks could be made by sending
```

**Review by nissimsan — APPROVED** [review]

**Pull request merged**